webhook/main.py

```diff
@@ -1,7 +1,24 @@
-from fastapi import FastAPI, Request
+from fastapi import FastAPI, Request, Header, HTTPException
 
 app = FastAPI()
 
+
 @app.get("/")
 async def root():
     return {"data": "Hello World"}
+
+
+@app.post("/query-results/")
+async def query_results(request: Request):
+    if request.headers.get("X-Appengine-Inbound-Appid", None) != 'virustotal-step-2020':
+        raise HTTPException(status_code=403, detail="Access forbidden")
+
+    try:
+        data = await request.json()
+    except RuntimeError:
```

Comment:
Why `RuntimeError`? Add a comment explaining why you're catching this specific exception.

Also, doesn't `pydantic` already checks the request body is JSON? What about using a pydantic model to check the fields coming in the JSON request (https://fastapi.tiangolo.com/tutorial/body/)?